cmd/cost/list.go

```diff
@@ -97,13 +94,56 @@ func (f listCostResponse) String() string {
 
 }
 
+type listAccountCostResponse struct {
+	AccountId string  `json:"accountid" yaml:"accountid"`
+	Unit      string  `json:"unit" yaml:"unit"`
+	Cost      float64 `json:"cost" yaml:"cost"`
+}
+
+func (f listAccountCostResponse) String() string {
+	return fmt.Sprintf("  AccountId: %s\n  Unit: %s\n  Cost: %f\n", f.AccountId, f.Unit, f.Cost)
+
+}
+
 func newListOptions(streams genericclioptions.IOStreams, globalOpts *globalflags.GlobalOptions) *listOptions {
 	return &listOptions{
 		IOStreams:     streams,
 		GlobalOptions: globalOpts,
 	}
 }
 
+func (ops *listOptions) runList() error {
+	awsClient, err := opsCost.initAWSClients()
+	cmdutil.CheckErr(err)
+
+	OU := getOU(awsClient, ops.ou)
+
+	var cost float64
+	var unit string
+	ouCost := OUCost{
+		OU:      OU,
+		options: ops,
+	}
+
+	if ops.level == "ou" {
+		if err := listCostsUnderOU(OU, awsClient, ops); err != nil {
+			log.Fatalln("Error listing costs under OU:", err)
+		}
+	} else {
+		ouCost.getCost(awsClient) // Get cost per account, print per account
+		if ops.level == "account" {
+			ouCost.printCostPerAccount()
+		}
+		cost, unit, err = ouCost.getSum() // Sum up account costs
+		if err != nil {
+			log.Fatalln("Error summing up cost of OU:", err)
+		}
+		printCostList(cost, unit, OU, ops, true) // TODO: Update bool here
```

Comment:
Why does this not have a return value?

Reply:
you're right, printCostList doesn't return an error, so we should update the function signature.